insights/client/connection.py

```diff
@@ -233,13 +233,14 @@ def user_agent(self):
         """
         Generates and returns a string suitable for use as a request user-agent
         """
-        import pkg_resources
-        core_version = "insights-core"
-        pkg = pkg_resources.working_set.find(pkg_resources.Requirement.parse(core_version))
-        if pkg is not None:
-            core_version = "%s %s" % (pkg.project_name, pkg.version)
-        else:
+        import importlib.metadata
```

Comment:
Hi. This branch still has to support Python 2.7, `importlib.metadata` have been [introduced in 3.8](https://docs.python.org/3/library/importlib.metadata.html#:~:text=Added%20in%20version), this won't work :/

Reply:
Hi, insights-core has only one branch which is targeting RHEL7/8/9/10, so your code needs to be in Python2.7

Reply:
Understood. Amedned to fallback to pkg_resources, as importlib_metadta might not be available in RHEL 7.

Happy to change the try-import to if sys.version_info check.